src/components/transactions/TxList/index.tsx

```diff
@@ -29,38 +32,59 @@ export const TxListGrid = ({ children }: { children: (ReactElement | null)[] }):
 
 const TxList = ({ items }: TxListProps): ReactElement => {
   const router = useRouter()
-  // Ensure list always starts with a date label
-  const list = useMemo(() => {
-    const firstDateLabelIndex = items.findIndex(isDateLabel)
-    const firstTxIndex = items.findIndex(isTransactionListItem)
-    const shouldPrependDateLabel =
-      (firstDateLabelIndex === -1 || firstDateLabelIndex > firstTxIndex) && firstTxIndex !== -1
 
-    if (!shouldPrependDateLabel) {
+  const list = useMemo(() => {
+    if (!hasTxFilterQuery(router.query)) {
       return items
     }
 
+    // Filtered transaction lists do not contain date labels
+    // Prepend initial date label to list
+    const firstTxIndex = items.findIndex(isTransactionListItem)
+
     const dateLabel: DateLabel = {
-      type: 'DATE_LABEL',
+      type: TransactionListItemType.DATE_LABEL,
       timestamp: (items[firstTxIndex] as Transaction).transaction.timestamp,
     }
+    const prependedItems = [dateLabel, ...items]
```

Comment:
I tried to achieve this with `items.slice().splice(0, 0, dateLabel)` but it was causing render issues.

Reply:
You mean `[dateLabel].concat(items)`?

Reply:
Nice touch! TypeScript isn't happy about that but I'll see if I can work some magic.